src/NUnitFramework/framework/Constraints/IndexerConstraint.cs

```diff
@@ -59,12 +59,22 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
         {
             Guard.ArgumentNotNull(actual, nameof(actual));
 
+            object indexedValue;
             var actualType = actual as Type ?? actual.GetType();
 
-            var getMethod = Reflect.GetDefaultIndexer(actualType, _argumentTypes) ?? throw new ArgumentException($"Default indexer accepting arguments {MsgUtils.FormatCollection(_arguments)} was not found on {actualType}."); ;
+            if (actualType.IsArray || typeof(Array).IsAssignableFrom(actualType))
+            {
+                var array = actual as Array;
+                indexedValue = array?.GetValue((int)_arguments[0]);
```

Comment:
Also, it would be great to check array rank and handle multi-dimensional arrays.

Reply:
Ah, in my haste I missed the API for:
```c#
public object GetValue(params int[] indices);
```
will update :+1: